rest/src/main/java/io/syndesis/rest/v1/handler/connection/ConnectionActionHandler.java

```diff
@@ -58,4 +84,42 @@ public ActionDefinition definition(@PathParam("id") @ApiParam(required = true,
         return action.map(Action::getDefinition).orElseThrow(() -> new EntityNotFoundException("Action with id: " + id));
     }
 
+    @GET
+    @Path(value = "/{id}/properties")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation("Retrieves action property suggestions")
+    @ApiResponses(@ApiResponse(code = 200, response = ActionPropertySuggestions.class,
+        message = "A map of zero or more action property suggestions keyed by the property name"))
+    public ActionPropertySuggestions properties(
+        @PathParam("id") @ApiParam(required = true,
+            example = "io.syndesis:salesforce-create-or-update:latest") final String id,
+        @Context final UriInfo uriInfo) {
+        final String connectorId = connector.getId().get();
+        final Action action = connector.actionById(id).orElseThrow(
+            () -> new EntityNotFoundException("No Action with id: " + id + ", found for connector: " + connectorId));
+
+        // put all specified parameters in query parameters
+        final MultivaluedMap<String, String> queryParameters = uriInfo.getQueryParameters();
+        final Map<String, Object> parameters = queryParameters.entrySet().stream()
+            .collect(Collectors.toMap(Entry::getKey, e -> e.getValue().get(0)));
+
+        // put all action parameters with `null` values
+        action.getDefinition().getPropertyDefinitionSteps()
+            .forEach(step -> step.getProperties().forEach((k, v) -> parameters.putIfAbsent(k, null)));
+
+        // lastly put all connection properties
+        parameters.putAll(connection.getConfiguredProperties());
+
+        final Client client = createClient();
+        final WebTarget target = client
+            .target(String.format("http://%s/api/v1/action/properties/%s", config.getService(), connectorId));
```

Comment:
Wonder whether we should change to https at some point. Same for the actual verifier connection.
Opened https://github.com/syndesisio/syndesis-rest/issues/565 for this.